### ISSUE TITLE
Garbled extraction for Amazon Sustainability Report

### ISSUE THREAD

**Issue by gtmtech** [pymupdf/PyMuPDF#3594]
### Description of the bug

Using PyMuPDF or PyMuPDF4LLM to extract the Amazon Sustainability Report gives quite incomprehensible output with beginnings of words garbled, and spacing screwed up. Something about this PDF is not being parsed properly. 

I have found other extractors do extract this report correctly.

### How to reproduce the bug

```
import fitz
d = fitz.open("file.pdf")
for p in d.pages()
    print(p.get_text())
```

[07ef2453.pdf](https://github.com/user-attachments/files/15898073/07ef2453.pdf)

### PyMuPDF version

1.24.2

### Operating system

MacOS

### Python version

3.11

**Comment by JorjMcKie** [comment]
[page0.pdf](https://github.com/user-attachments/files/15901270/page0.pdf)
[page0-1.pdf](https://github.com/user-attachments/files/15901271/page0-1.pdf)

**Comment by JorjMcKie** [comment]
The Amazon font(s) contain non-standard encodings. In such cases, the success of text extraction is subject to good luck.
Some extractors may be better than others in guessing what the right Unicode for a given glyph may be.

The weird characters stem from the default flag bit `TEXT_CID_FOR_UNKNOWN_UNICODE`. When on, MuPDF interprets the extracted glyph number as the Unicode of a character.
This often helps in these problem cases - but not always, like here.
You can set it off - either completely with all other flag bits via `flags=0` or selectively via e.g. `flags=pymupdf.TEXTFLAGS_TEXT & ~pymupdf.TEXT_CID_FOR_UNKNOWN_UNICODE`.
Any non-existing Unicode will then be reported as "&#xfffd;".

I also have submitted a question to the MuPDF team for any other advices they may have.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.6.
